### PR TITLE
[CELEBORN-1490][CIP-6] Add Flink Hybrid Shuffle IT test cases

### PR DESCRIPTION
**Pull request by reswqa**
### What changes were proposed in this pull request?
1. Add Flink Hybrid Shuffle IT test cases
2. Fix bug in open stream.

### Why are the changes needed?

Test coverage for celeborn + hybrid shuffle

### Does this PR introduce _any_ user-facing change?
No

### CODE REVIEW

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/readclient/CelebornBufferStream.java [hunk 1]
```diff
@@ -312,8 +323,19 @@ public void onFailure(Throwable e) {
...
             messageConsumer.accept(new TransportableError(streamId, e));
+            if (resultFuture != null) {
+              resultFuture.set(null);
+            }
           }
         });
+
+    if (sync) {
+      try {
+        resultFuture.get();
```
Comment (reswqa) [inline client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/readclient/CelebornBufferStream.java:334]: We should `openStream` synchronized as `requiredSegmentId` RPC have to be sent after open.

**Review by SteNicholas — APPROVED** [review]

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/readclient/CelebornBufferStream.java [hunk 2]
```diff
@@ -255,6 +256,12 @@ private void openStreamInternal(@Nullable BiConsumer<Long, Integer> requiredSegm
                 .setInitialCredit(initialCredit)
                 .build()
                 .toByteArray());
+    final SettableFuture<Void> resultFuture;
```
Comment (RexXiong) [inline client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/readclient/CelebornBufferStream.java:259]: Could we create a new method in TransportClient for the sync rpc?

`public ByteBuffer sendRpcSync(ByteBuffer message, RpcResponseCallback callback, long timeoutMs)`

**Review by RexXiong — APPROVED** [review]
LGTM, thanks

**Comment by SteNicholas** [comment]
Thanks. Merged to main(v0.6.0).

**Pull request closed without merging**